modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/AbstractParameterResolverExecutor.java

```diff
@@ -87,9 +93,13 @@ protected ParameterValueResolver parameterValueResolver(ParameterizedElementDecl
         final ParameterModel parameterModel = parameterGroupModel.getParameter(parameterName)
             .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find parameter with name: '%s' in parameter group: '%s'",
                                                                             parameterName, parameterGroupName)));
-        parametersMap.put(parameterName,
-                          extractValue(parameterElement.getValue(),
-                                       artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration)));
+        Object value = extractValue(parameterElement.getValue(),
+                                    artifactHelper.getParameterClass(parameterModel, parameterizedElementDeclaration));
+        if (isExpression(value)) {
```

Comment:
I just realized that I should first check if the parameterModel.isSupportExpression() == true before checking the String content.